src/models/os.ts

```diff
@@ -384,17 +397,38 @@ const getOsModel = function (
 	 * @example
 	 * balena.models.os.getAvailableOsVersions(['fincm3', 'raspberrypi3']);
 	 */
-	async function getAvailableOsVersions(
+	async function getAvailableOsVersions<
+		TP extends PineOptions<Release> | undefined,
+	>(
 		deviceTypes: string[] | string,
-		options?: { includeDraft?: boolean },
-	): Promise<TypeOrDictionary<OsVersion[]>> {
+		// TODO: Consider providing a different way to for specifying includeDraft in the next major
+		// eg: make a methods that returns the complex filter
+		options?: TP & { includeDraft?: boolean },
+	): Promise<
+		TypeOrDictionary<Array<ExtendedPineTypedResult<Release, OsVersion, TP>>>
+	> {
+		const pineOptionEntries =
+			options != null
+				? Object.entries(options).filter(([key]) => key.startsWith('$'))
+				: undefined;
+		const pineOptions =
+			pineOptionEntries != null && pineOptionEntries.length > 0
```

Comment:
nit
```suggestion
			!!pineOptionEntries?.length
```

Reply:
I would actually prefer to be more explicit than short :)